packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -118,12 +118,31 @@ export class MediaPlaceholder extends Component {
 			onError,
 			onSelect,
 			value = [],
+			dynamicAlteration,
 		} = this.props;
 		let setMedia;
 		if ( multiple ) {
 			if ( addToGallery ) {
-				const currentValue = value;
+				let currentValue = value;
 				setMedia = ( newMedia ) => {
+					if ( dynamicAlteration ) {
+						const galleryValue = this.props.value || [];
+						// If currentValue contains things not in galleryValue, remove them.
+						currentValue = currentValue.filter( ( item ) => {
+							return galleryValue.some( ( propItem ) => propItem.id === item.id && propItem.url === item.url );
+						} );
+						// If galleryValue has completed items not in currentValue, add them.
+						galleryValue.forEach( ( propItem ) => {
```

Comment:
Similarly to above,  what if instead of `propItem`, we said `newItem`?

Reply:
I updated it to be `currentItem` to fit with `currentMedia` and `mediaToReturn`, although `currentItem` still seems kind of odd it does at least make sense being an item from `currentMedia`.